### PR TITLE
Feature/Convertors

### PR DESCRIPTION
**Pull request by lilbonekit**

### CODE REVIEW

File: assets/icons/constant-icon.svg
Comment (KlausKidman) [inline assets/icons/constant-icon.svg:1]: rename to `square-root.svg`

File: enums/constants.enum.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export enum BYTES_CONSTANTS {
+  zeroBytes = '0x0000000000000000000000000000000000000000000000000000000000000000',
+}
```
Comment (KlausKidman) [inline enums/constants.enum.ts:3]: need to discuss

File: enums/icon-names.enum.ts [hunk 2]
```diff
@@ -21,4 +21,5 @@ export enum ICON_NAMES {
   refresh = 'refresh',
   xCircle = 'x-circle',
   x = 'x',
+  constant = 'constant',
```
Comment (KlausKidman) [inline enums/icon-names.enum.ts:24]: !

File: forms/AsciiConverterForm.vue [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        :model-value="value"
+        :key="key"
+        :label="$t(`ascii-converter-form.${key}-label`)"
+        :placeholder="$t(`ascii-converter-form.${key}-placeholder`)"
+        :error-message="getFieldErrorMessage(key)"
+        @update:model-value="formatInputs($event, key)"
+        @blur="touchField(key)"
+      >
+        <template #nodeLeft>
+          <app-copy :value="form[key] || 0" />
```
Comment (KlausKidman) [inline forms/AsciiConverterForm.vue:17]: `form[key]` -> `value`

File: forms/AsciiConverterForm.vue [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+  switch (key) {
+    case 'hexadecimal':
+      form.ascii = hexToASCII(form.hexadecimal)
+      break
+    case 'ascii':
+      form.hexadecimal = asciiToHex(form.ascii)
+      break
+  }
+}
```
Comment (KlausKidman) [inline forms/AsciiConverterForm.vue:73]: need to discuss

File: forms/ConstantsForm.vue [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+            >
+              <app-icon
+                class="constants-form__btn-icon"
+                :name="$icons.refresh"
+              />
+            </button>
+          </div>
+          <input-field
+            readonly
+            :model-value="form[key]"
```
Comment (KlausKidman) [inline forms/ConstantsForm.vue:26]: !

File: forms/ConstantsForm.vue [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+const generateRandomBytes = (): string => hexlify(randomBytes(32))
+
+const form = reactive({
+  zeroBytes32: BYTES_CONSTANTS.zeroBytes,
+  randomBytes32: generateRandomBytes(),
+})
+
+onMounted(() => {
+  form.randomBytes32 = generateRandomBytes()
+})
```
Comment (KlausKidman) [inline forms/ConstantsForm.vue:67]: for what?

File: forms/DurationForm.vue [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+  return getTotalDurationAsSeconds(rawDuration)
+})
+
+watch(
+  () => form.duration,
+  () => {
+    Object.assign(rawDuration, INITIAL_RAW_DURATION_STATE)
+    Object.assign(rawDuration, getUpdatedDuration(form.duration, rawDuration))
+  },
+)
```
Comment (KlausKidman) [inline forms/DurationForm.vue:105]: need to discuss

File: forms/VerifySignatureForm.vue [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+        />
+        <div class="verify-signature-form__buttons">
+          <app-button
+            type="submit"
+            :text="$t('verify-signature-form.submit-btn')"
+          />
+          <app-button
+            scheme="flat"
+            :text="$t('verify-signature-form.reset-btn')"
+            @click="resetForm()"
```
Comment (KlausKidman) [inline forms/VerifySignatureForm.vue:35]: `@click="resetForm"`

File: forms/ConstantsForm.vue [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+
+<script lang="ts" setup>
+import { AppCopy, AppIcon } from '#components'
+import { InputField } from '@/fields'
+import { v4 as uuidv4 } from 'uuid'
+import { onMounted, reactive } from 'vue'
+import { randomBytes, hexlify } from 'ethers'
+import { BYTES_CONSTANTS } from '@/enums'
+
+const randomBytes32InputFieldUid = `input-field--${uuidv4()}`
```
Comment (KlausKidman) [inline forms/ConstantsForm.vue:57]: need to discuss

File: forms/DurationForm.vue [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+const durationObject = computed(() => {
+  const durationObject: Partial<Record<keyof RawDateType, number>> = {}
+  const durationKeys = Object.keys(rawDuration) as (keyof RawDateType)[]
+
+  durationKeys.forEach(key => {
+    durationObject[key] = rawDuration[key]
+  })
+
+  return durationObject
+})
```
Comment (KlausKidman) [inline forms/DurationForm.vue:89]: !

File: helpers/time.helpers.ts [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+
+  let match
+  while ((match = regex.exec(dateString))) {
+    const value = parseInt(match[1])
+    const unit = match[2] as TIME_IDS
+
+    if (unitMap[unit]) {
+      updatedDuration[unitMap[unit]] += value
+    }
+  }
```
Comment (KlausKidman) [inline helpers/time.helpers.ts:29]: ?

File: forms/UnitConverterForm.vue [hunk 12]
```diff
@@ -2,20 +2,21 @@
...
         :label="UNITS[key].title"
         :placeholder="UNITS[key].title"
-        :model-value="form[key]"
         :error-message="getFieldErrorMessage(key)"
         @blur="touchField(key)"
         @update:model-value="formatInputs($event, key)"
       >
         <template #nodeLeft>
-          <app-copy :value="form[key] || 0" />
+          <app-copy :value="value || 0" />
```
Comment (lilbonekit) [inline forms/UnitConverterForm.vue:17]: To discuss again
@KlausKidman

**Review by KlausKidman — APPROVED** [review]

**Pull request merged**